integration/tests/basic/erc/test_ERC20SPLnew.py

```diff
@@ -0,0 +1,1221 @@
+
+    @pytest.mark.skip("https://neonlabs.atlassian.net/browse/NDEV-3052")
+    def test_update_factory_invalid_owner(self, erc20_spl_mintable, accounts, new_factory_contract):
+        factory_contract = self.get_factory_contract(erc20_spl_mintable.contract)
+        tx_opt = self.web3_client.make_raw_tx(accounts[0], gas=10000000)
+        tx_body = factory_contract.functions.upgradeToAndCall(new_factory_contract.address, "0x").build_transaction(
+            tx_opt
+        )
+        receipt = self.web3_client.send_transaction(accounts[0], tx_body)
+        assert receipt.status == 0, "Transaction should fail"
+
+    def test_update_token_contract(self, erc20_spl_mintable_new, new_token_contract):
+        # get contract factory object, get address from erc20 token contract
+        factory_contract = self.get_factory_contract(erc20_spl_mintable_new.contract)
+        # update factory implementation
+        impl_address_before_update = self.web3_client.eth.get_storage_at(factory_contract.address, 0).hex()
+        tx_opt = self.web3_client.make_raw_tx(erc20_spl_mintable_new.account)
+        tx_body = factory_contract.functions.upgradeTo(new_token_contract.address).build_transaction(tx_opt)
+        receipt = self.web3_client.send_transaction(erc20_spl_mintable_new.account, tx_body)
+        assert receipt.status == 1, "Transaction failed"
+
+        # check proxy factory address
+        impl_address_after_update = self.web3_client.eth.get_storage_at(factory_contract.address, 0).hex()
+        assert impl_address_before_update != impl_address_after_update, "Token implementation wasn't updated"
+
+        # deploy new token
+        deploy_tx = self.web3_client.make_raw_tx(erc20_spl_mintable_new.account)
+        deploy_body = factory_contract.functions.deploy("TOKEN3", "ABC", "http://uri2.com", 8).build_transaction(
+            deploy_tx
+        )
+        deploy_receipt = self.web3_client.send_transaction(erc20_spl_mintable_new.account, deploy_body)
+        logs = factory_contract.events.TokenDeploy().process_receipt(deploy_receipt)
+        new_token_addr = logs[0]["args"]["token"]
+        new_token_contract = self.web3_client.get_deployed_contract(
+            new_token_addr,
+            contract_file="external/neon-contracts/ERC20ForSPL/contracts/test/ERC20ForSPLMintableV2",
+            solc_version="0.8.24",
+        )
+        assert new_token_contract.functions.name().call() == "TOKEN3"
+        assert new_token_contract.functions.getDummyData().call() == 1112131415
+
+
+@allure.feature("ERC Verifications")
+@allure.story("ERC20SPL: Tests for ERC20ForSPL contract")
+@pytest.mark.usefixtures("accounts", "web3_client", "sol_client")
+@pytest.mark.neon_only
+class TestERC20SPLNewFeatures:
+    web3_client: NeonChainWeb3Client
+    accounts: EthAccounts
+    sol_client: SolanaClient
+
+    @pytest.fixture(scope="class")
+    def erc20_contract(self, erc20_spl_new, eth_bank_account, pytestconfig: Config) -> ERC20NewWrapper:
```

Comment:
this fixture looks strange 
why do you need send_neon for mainnet ( you can't have free neon on mainnet)